### PR TITLE
Ft refactor

### PR DESCRIPTION
**Pull request by Collins48**

### CODE REVIEW

File: src/main/java/com/collicode/propertytracker/service/spec/PaymentsService.java [hunk 1]
```diff
@@ -8,4 +8,5 @@ public interface PaymentsService {
   void createPayment(PaymentRequestDTO paymentRequestDTO);
   List<PaymentResponseDTO> fetchPaymentByTenantId(long tenantId);
 
+  Object deletePayment(PaymentRequestDTO paymentId);
```
Comment (c-kiplimo) [inline src/main/java/com/collicode/propertytracker/service/spec/PaymentsService.java]: It should hae a return type may be String also while delete it should not take an object 
 example 

`String deletePaymentById(String paymentId);`           but we should not delete payment actually as we need to keep records.

File: src/main/java/com/collicode/propertytracker/service/impl/TenantsServiceImpl.java [hunk 2]
```diff
@@ -61,6 +65,18 @@ public void createTenant(TenantRequestDTO tenantRequestDTO) {
           throw StorageException.exception("ERROR WHILE CREATING TENANT");
         }
     }
+    @DeleteMapping("/{tenantId}")
+    public ResponseEntity<?> deleteTenant(@RequestHeader("X-RequestId") String requestId,
+                                             @PathVariable long tenantId) {
+        try {
+            return ResponseEntity.ok().body(TenantRepository.deleteTenant(tenantId));
```
Comment (c-kiplimo) [inline src/main/java/com/collicode/propertytracker/service/impl/TenantsServiceImpl.java]: Controler should not interact with the repository directly ,it should be through a service.

Comment (Collins48) [reply src/main/java/com/collicode/propertytracker/service/impl/TenantsServiceImpl.java]: noted

File: src/main/java/com/collicode/propertytracker/service/spec/PaymentsService.java (hunk 1 above)
Comment (Collins48) [reply src/main/java/com/collicode/propertytracker/service/spec/PaymentsService.java]: so should i not create the crud for deleting payments

Comment (c-kiplimo) [reply src/main/java/com/collicode/propertytracker/service/spec/PaymentsService.java]: Yes.

**Comment by Collins48** [comment]
in the matter crud pr i have added the delete crud for the tenants the PUT crude for the payments and made the changes

File: src/main/java/com/collicode/propertytracker/api/TenantApiController.java [hunk 3]
```diff
@@ -35,5 +36,15 @@ public  ResponseEntity<?> fetchTenants(@RequestHeader("X-RequestId") String requ
             return ResponseEntity.status(500).body(ex.getMessage());
         }
     }
-
+    @DeleteMapping("/{tenantId}")
+    public ResponseEntity<?> deleteTenant(@RequestHeader("X-RequestId") String requestId,
+                                          @PathVariable long tenantId) {
+        try {
+            return ResponseEntity.ok().body(TenantService.deleteTenant(tenantId));
```
Comment (c-kiplimo) [inline src/main/java/com/collicode/propertytracker/api/TenantApiController.java]: Autowire the class

```
@Autowire
private TenantService tenantService

tenantService.deleteTenant(tenantId);
```

**Review by c-kiplimo — APPROVED** [review]

**Pull request merged**